yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala

```diff
@@ -38,24 +38,30 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
   var amMemory: Int = 512 // MB
   var appName: String = "Spark"
   var priority = 0
+  var isClusterMode = false
+
+  private val driverMemKey = "spark.driver.memory"
+  private val driverMemOverheadKey = "spark.yarn.driver.memoryOverhead"
+  private val amMemKey = "spark.yarn.am.memory"
+  private val amMemOverheadKey = "spark.yarn.am.memoryOverhead"
+  private var isDriverMemSet = false
 
   parseArgs(args.toList)
 
+  private val isDynamicAllocationEnabled =
+    sparkConf.getBoolean("spark.dynamicAllocation.enabled", false)
```

Comment:
I would prefer you move this up there with other `private vals` and keep the calls to `parseArgs`, `loadEnvironmentArgs` and `validateArgs` together